### PR TITLE
Updated classes to add new fields per hip 1075.

### LINKED ISSUE #1075: HIP-423 Enhance Scheduled Transactions Support
### Problem

The existing schedule transactions feature is being enhanced. Please refer to HIP-423 for details.

HIP

https://hips.hedera.com/hip/hip-423

Protobufs

https://github.com/hashgraph/hedera-protobufs/pull/184
https://github.com/hashgraph/hedera-protobufs/pull/186

### Solution

Add support

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by keirabee**
Signed-off-by: Keira Black <keirablack@Keiras-MacBook-Pro.local>

**Description**:

**Related issue(s)**:

Fixes #1075 

**Notes for reviewer**:

**Checklist**

- [X] Documented (Code comments, README, etc.)
- [X] Tested (unit, integration, etc.)

### CODE REVIEW

File: src/schedule/ScheduleCreateTransaction.js [hunk 1]
```diff
@@ -298,6 +312,41 @@ export default class ScheduleCreateTransaction extends Transaction {
...
+    get expirationTime() {
+        this._requireNotFrozen();
+        return this._expirationTime;
+    }
+
+    /**
+     * @param {boolean} waitForQuery
+     * @returns {this}
+     */
+    _setWaitForQuery(waitForQuery) {
```
Comment (danielakhterov) [inline src/schedule/ScheduleCreateTransaction.js:337]: Should not be prefixed with an `_`

File: src/schedule/ScheduleCreateTransaction.js [hunk 2]
```diff
@@ -298,6 +312,41 @@ export default class ScheduleCreateTransaction extends Transaction {
         );
         return `ScheduleCreateTransaction:${timestamp.toString()}`;
     }
+
+    /**
+     * @param {?Timestamp} expirationTime
+     * @returns {this}
+     */
+    _setExpirationTime(expirationTime) {
```
Comment (danielakhterov) [inline src/schedule/ScheduleCreateTransaction.js:320]: Should not be prefixed with an `_`

File: src/schedule/ScheduleInfoQuery.js [hunk 3]
```diff
@@ -206,6 +212,23 @@ export default class ScheduleInfoQuery extends Query {
 
         return `ScheduleInfoQuery:${timestamp.toString()}`;
     }
+
+    /**
+     * @param {boolean} waitForQuery
+     * @returns {this}
+     */
+    _setWaitForQuery(waitForQuery) {
```
Comment (danielakhterov) [inline src/schedule/ScheduleInfoQuery.js:220]: Should not be prefixed with an `_`

File: src/schedule/ScheduleInfoQuery.js [hunk 4]
```diff
@@ -206,6 +212,23 @@ export default class ScheduleInfoQuery extends Query {
 
         return `ScheduleInfoQuery:${timestamp.toString()}`;
     }
+
```
Comment (danielakhterov) [inline src/schedule/ScheduleInfoQuery.js:215]: I think you need to update `[to|from]Protobuf()` methods to support this new field.

File: src/schedule/ScheduleCreateTransaction.js [hunk 5]
```diff
@@ -298,6 +312,41 @@ export default class ScheduleCreateTransaction extends Transaction {
...
+        return this;
+    }
+
+    /**
+     * @returns {?boolean}
+     */
+    get waitForQuery() {
+        this._requireNotFrozen();
+        return this._waitForQuery;
+    }
```
Comment (danielakhterov) [inline src/schedule/ScheduleCreateTransaction.js:349]: I think you need to update the `[to|from]Protobuf()` methods to support  this new field.

**Review by danielakhterov — CHANGES_REQUESTED** [review]

File: src/schedule/ScheduleCreateTransaction.js (hunk 1 above)
Comment (keirabee) [reply src/schedule/ScheduleCreateTransaction.js:337]: fixed.

File: src/schedule/ScheduleCreateTransaction.js (hunk 2 above)
Comment (keirabee) [reply src/schedule/ScheduleCreateTransaction.js:320]: fixed.

File: src/schedule/ScheduleInfoQuery.js (hunk 3 above)
Comment (keirabee) [reply src/schedule/ScheduleInfoQuery.js:221]: fixed.

File: src/schedule/ScheduleInfoQuery.js (hunk 4 above)
Comment (keirabee) [reply src/schedule/ScheduleInfoQuery.js:216]: updated to include field.

File: src/schedule/ScheduleCreateTransaction.js (hunk 5 above)
Comment (keirabee) [reply src/schedule/ScheduleCreateTransaction.js:349]: updated to include field.

File: src/schedule/ScheduleCreateTransaction.js (hunk 2 above)
Comment (danielakhterov) [reply src/schedule/ScheduleCreateTransaction.js:320]: I think you forgot to push :/

**Comment by keirabee** [comment]
Closing this out as the relevant changes are now in another pull request, #1122. https://github.com/hashgraph/hedera-sdk-js/pull/1122

**Pull request closed without merging**